target/scripts/build/packages.sh

```diff
@@ -1,10 +1,13 @@
 #!/bin/bash
 
-# -eE :: exit on error (do this in functions as well)
-# -u  :: show (and exit) when using unset variables
+# -eE         :: exit on error (do this in functions as well)
```

Comment:
Some of the changes in `packages.sh` here could probably be split out to a separate PR as it's not all related to the image upgrade.

The refactoring probably has been causing various conflicts to resolve too. Not a major concern anymore since this PR will finally get merged soon after final review :)

Reply:
Yeah, somehow typical for my PRs haha. Let's merge this PR anyway; reverting it now is something I can't affort time-wise.

Reply:
BTW: `-E` is not "exit on error in functions". `-e` also applies to functions.

```-E      If set, any trap on ERR is inherited by shell functions, command substitutions, and commands executed in a sub‐ shell environment. The ERR trap is normally not inher‐ ited in such cases.```

For the future, we could consider to add `shopt -s inherit_errexit` with applies `-e` to subshells.

Reply:
@casperklein what would be a better wording for the doc comment then? :)

Reply:
`-E      inherit the ERR trap to functions, command substitutions and sub-shells`

Off-Topic: I've updated docker to version 25.0.0 and noticed, that the init script was changed. I checked the docker repo and a guy whose username sounds familiar to me added `ulimit -Hn 524288` 😆

Reply:
> Off-Topic: I've updated docker to version 25.0.0 and noticed, that the init script was changed. I checked the docker repo and a guy whose username sounds familiar to me added `ulimit -Hn 524288` 😆

Link please :D

Reply:
See [here](https://github.com/moby/moby/pull/45534/files#diff-329a7d3594b1ed91f76deded4734c7d9390c4dce74aeaa5a772fbf52b0a58c2e).

Reply:
> I checked the docker repo and a guy whose username sounds familiar to me added `ulimit -Hn 524288` 😆

Yeaaah... it was well meant but apparently I made some assumptions on compatibility (_although in some cases, it [turns out the previous `ulimit` command and value could be just a no-op](https://github.com/docker/cli/issues/4807#issuecomment-1903950217)_), rather than serving the intended purpose and this change just surfaced that issue for those affected.

It's worse with containerd where my change is getting backlash from business users for breaking their production setups. I still think matching host limits by default is the right decision, but it's ruffled some feathers 😬

Reply:
Not sure if this was what @casperklein had in mind 🤷‍♂️ 

```suggestion
# -e          :: exit on error (do this in functions as well)
# -E          :: inherit the ERR trap to functions, command substitutions and sub-shells
```